modern/src/ReloadPrompt.tsx

```diff
@@ -0,0 +1,56 @@
+import { Snackbar, Button } from '@mui/material';
+import React from 'react'
+import { useTranslation } from './common/components/LocalizationProvider';
+import { useAttributePreference } from './common/util/preferences';
+import { useRegisterSW } from 'virtual:pwa-register/react'
+
+// Based on https://vite-pwa-org.netlify.app/frameworks/react.html
+function ReloadPrompt() {
```

Comment:
We need a better name because it's not just prompt. It actually checks.

Reply:
UpdateCheckPrompt ? or even UpdateCheckAndPrompt?

Reply:
Went with UpdateCheckPrompt for now, after looking around UpdateController could be used too. What do you think?